chore: replace jest with vitest

hereje opened the pull request:
Replace jest with vitest:

- add vitest
- remove jest
- add vitest api globally
- update test files
- replace jest config with vitest config
- solve integration issues
- solve typo issue
Related to: #20

hereje committed:
feat: replace jest with vitest

- add vitest
- remove jest
- add vitest api globally
- update test files
- replace jest config with vitest config

hereje committed:
fix: exclude vitest config file from rootDir

- update tsconfig.json config so that vitest config file does not
mess up with the current data type generation

hereje committed:
fix: update dependencies running npm audit fix

malinovic commented on package.json:
```diff
@@ -54,11 +53,11 @@
     "terminal-link": "2.1.1",
-    "ts-config-webpack-plugin": "2.0.3",
-    "ts-jest": "29.1.2",
+    "ts-config-webpack-plugin": "2.0.0",
```
Why moving ts-config-webpack-plugin from 2.0.3 to 2.0.0?

hereje replied:
I've run `npm audit fix`

malinovic commented on package.json:
```diff
@@ -54,11 +53,11 @@
     "typescript": "5.3.3",
-    "webpack": "4.47.0",
+    "vitest": "3.2.4",
+    "webpack": "5.101.3",
```
I think that bumping webpack from 4 to 5 is a bad idea at the moment. Since it is still working with webpack 4...

hereje replied:
OK, I'll revert it.

malinovic reviewed (changes requested):
I have checked your code. Testing seems fine to me. I have tried to build the project locally and the build fails right now. 
Please take a look on the comments I have left and check the failing build. 
Thank you.

hereje committed:
fix: revert webpack upgrade to 4.47.0

- revert webpack upgrade to 4.47.0
- revert ts-config-webpack-plugin downgrade to 2.0.3
